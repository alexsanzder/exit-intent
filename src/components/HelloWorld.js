import React from 'react';

const HelloWorld = ({ name }) => <h1>{`Hello ${name}!`}</h1>;

export default HelloWorld;
