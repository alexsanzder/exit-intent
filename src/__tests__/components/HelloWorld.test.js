import React from 'react';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';
import HelloWorld from '../../components/HelloWorld';

describe(HelloWorld, () => {
  // Add the rest of our tests here later!
  const name = 'Person';
  const component = shallow(<HelloWorld name={name} />);

  it('renders and matches our snapshot', () => {
    const component = renderer.create(<HelloWorld name="Person" />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('contains the supplied name', () => {
    expect(component.text()).toContain(name);
  });
});
