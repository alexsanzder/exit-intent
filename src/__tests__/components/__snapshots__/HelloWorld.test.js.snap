// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`({ name }) => _react2.default.createElement('h1', { __source: { fileName: _jsxFileName, lineNumber: 3 }, __self: undefined }, \`Hello \${name}!\`) renders and matches our snapshot 1`] = `
<h1>
  Hello Person!
</h1>
`;
